### ISSUE TITLE
magit status doesn't trigger a direnv update

### ISSUE THREAD

**Issue by stribb** [wbolster/emacs-direnv#48]
My setup:
`(setq projectile-switch-project-action 'magit-status)`

When (effectively) I run this:

`(projectile-switch-project-by-name "myproject")`

it would help for direnv to set the environment based on the project directory.

I notice that `(direnv--directory) => nil` in the context of magit-status buffers. Does it make sense for me to add `magit-mode` to `direnv-non-file-modes`? If so, may I propose to change `direnv--directory` thus:

```elisp
(defun direnv--directory ()
  "Return the relevant directory for the current buffer, or nil."
  (let* ((buffer (or (buffer-base-buffer) (current-buffer)))
         (file-name (buffer-file-name buffer)))
    (cond (file-name (file-name-directory file-name))
          ((apply derived-mode-p direnv-non-file-modes) default-directory))))
```

This way, any magit mode will be counted as a `dired-non-file-modes` member.

**Comment by wbolster** [comment]
yes, all makes sense! pr welcome for the derived-mode-p check

**Comment by stribb** [comment]
My code had a bug to do with indirect buffers, which I fix in #51. Once that's submitted I think we can close this issue.
